src/app/icd/server/ICDManager.h

```diff
@@ -71,11 +73,14 @@ class ICDManager : public ICDListener, public TestEventTriggerHandler
         ActiveMode,
     };
 
-    // This enum class represents to all ICDStateObserver callbacks available from the
-    // mStateObserverPool for the ICDManager.
+    /**
+     * @brief This enum class represents to all ICDStateObserver callbacks available from the
+     *        mStateObserverPool for the ICDManager.
+     */
     enum class ObserverEventType : uint8_t
     {
         EnterActiveMode,
+        EnterIdleMode,
         TransitionToIdle,
```

Comment:
The difference between EnterIdleMode and TransitionToIdle is extremely non-obvious.  Should be documented.

Similarly, how does ICDModeChange interact with the other events?  Instead of them?  In addition to them?  Something else?  Why does it exist at all?